### PR TITLE
Adds EKSConfig reconciliation integration tests

### PR DESCRIPTION
**Pull request by NicoleYson**
Adds EKSConfig reconciliation integration tests

### CODE REVIEW

File: bootstrap/eks/controllers/eksconfig_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/*
```
Comment (rudoi) [inline bootstrap/eks/controllers/eksconfig_controller_test.go:1]: WDYT about using the pattern in CAPI where the reconciliation tests are in a file called `eksconfig_controller_reconciler_tests.go` and leaving this filename for the unit tests?

File: bootstrap/eks/controllers/eksconfig_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+			APIVersion: clusterv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      name,
+		},
+	}
+}
+
+// newEKSConfig return a CABPK EKSConfig object; if machine is not nil, the EKSConfig is linked to the machine as well
```
Comment (rudoi) [inline bootstrap/eks/controllers/eksconfig_controller_test.go:91]: nit - "CABPK" refers specifically to the kubeadm provider, so it's a touch confusing in this comment

**Comment by rudoi** [comment]
@NicoleYson I think if we point this at `eks-bootstrap` for now we can merge it in ASAP, would hate to hold it up on the Makefile stuff.

File: bootstrap/eks/controllers/eksconfig_controller_reconciler_tests.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/test/helpers"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	testEnv *helpers.TestEnvironment
```
Comment (h0tbird) [inline bootstrap/eks/controllers/eksconfig_controller_reconciler_tests.go:36]: `testEnv` is already declared in: https://github.com/newrelic-forks/cluster-api-provider-aws/blob/93384b83e9def30779ea4940675f4997cb3b74bf/bootstrap/eks/controllers/suite_test.go#L42
I am think: remove this one and change the one in `suite_test.go` to look like `testEnv *helpers.TestEnvironment` (same approach as in [kubeadm](https://github.com/kubernetes-sigs/cluster-api/blob/4e07b0c23f6782fa5ca3729c3f9ed005a6f9a203/controlplane/kubeadm/controllers/suite_test.go#L34))

Comment (NicoleYson) [reply bootstrap/eks/controllers/eksconfig_controller_reconciler_tests.go:36]: 😮 I didn't notice it in `suite_test.go`
Will resolve this. Interesting that running the test cases for me did not call out the redeclaration 🤔 

Thanks @h0tbird !!!

**Comment by NicoleYson** [comment]
@h0tbird I was using `make test` 
I am still getting 1 error though with the new changes, so this PR isn't quite ready yet:
```
    Expected success, but got an error:
        <*runtime.notRegisteredErr | 0xc0007e4420>: {
            schemeName: "pkg/runtime/scheme.go:101",
            gvk: {Group: "", Version: "", Kind: ""},
            target: nil,
            t: {size: 0x170, ptrdata: 0x160, hash: 1061614163, tflag: 7, align: 8, fieldAlign: 8, kind: 25, equal: nil, gcdata: 85, str: 181272, ptrToThis: 3030304},
        }
        no kind is registered for the type v1alpha3.EKSConfig in scheme "pkg/runtime/scheme.go:101"
``` 

I'm hoping to resolve it soon 😄 🤞

**Pull request merged**